client/state/selectors/get-google-my-business-locations.js

```diff
@@ -1,50 +1,33 @@
 /**
  * External dependencies
  */
+import { filter } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import { getSiteSettings } from 'state/site-settings/selectors';
-import { getKeyringConnectionsByName } from 'state/sharing/keyring/selectors';
+import { getAvailableExternalAccounts } from 'state/sharing/selectors';
 
-function isConnected( keyringConnection, externalUser, siteSettings ) {
-	return keyringConnection.ID === siteSettings.google_my_business_keyring_id &&
-		externalUser.external_ID === siteSettings.google_my_business_location_id;
+function isConnected( externalUser, siteSettings ) {
+	return externalUser.keyringConnectionId === siteSettings.google_my_business_keyring_id &&
+		externalUser.ID === siteSettings.google_my_business_location_id;
 }
 
-/**
- * Returns a list of "external users" for the Google My Business keyring connection.
- * Those external users are Business Locations associated with this Google account
- * in this context
- *
- * @param  {Object} state  Global state tree
- * @param  {Object} siteId The site ID
- * @return {Object}        List of GMB locations
- */
 export default function getGoogleMyBusinessLocations( state, siteId ) {
 	const siteSettings = getSiteSettings( state, siteId );
 
 	if ( ! siteSettings ) {
 		return [];
 	}
 
-	const keyringConnections = getKeyringConnectionsByName( state, 'google_my_business' );
-	const locations = [];
+	const externalUsers = filter( getAvailableExternalAccounts( state, 'google_my_business' ), {
+		isExternal: true,
+	} );
 
-	keyringConnections.forEach( keyringConnection => {
-		if ( keyringConnection.additional_external_users ) {
-			keyringConnection.additional_external_users.forEach( externalUser => {
-				locations.push( {
-					...keyringConnection,
-					keyring_connection_ID: keyringConnection.ID,
-					external_ID: externalUser.external_ID,
-					external_display: externalUser.external_name,
-					isConnected: isConnected( keyringConnection, externalUser, siteSettings ),
-				} );
-			} );
-		}
+	externalUsers.forEach( externalUser => {
+		externalUser.isConnected = isConnected( externalUser, siteSettings );
 	} );
 
-	return locations;
+	return externalUsers;
 }
```

Comment:
I really don't think it's a good idea to return Keyring data from a function named `getGoogleMyBusinessLocations()`. The fact that we're using Keyring and site settings to manage these locations should be an implementation detail, and be hidden. I would have expected it to return a very simple object like so:

```
{
  id: 12345,
  address: [
    'Centre Commercial Cap 3000',
    'Avenue Eugene Donadei',
    '06700 Saint-Laurent-du-Var',
    'France',
  ],
  name: 'Starbucks',
  photo: 'http://www....jpg',
  verified: true,
}
```

Reply:
That's not the only place, we're leaking the fact that location is "externalAccount" further,
for example:
https://github.com/Automattic/wp-calypso/pull/23741/files#diff-7052b7b1fe065e12121838e62f6f7f41R128